openpype/lib/vendor_bin_utils.py

```diff
@@ -3,9 +3,65 @@
 import json
 import platform
 import subprocess
-import distutils
 
-log = logging.getLogger("FFmpeg utils")
+log = logging.getLogger("Vendor utils")
+
+
+def find_executable(executable):
+    """Find full path to executable.
+
+    Also tries additional extensions if passed executable does not contain one.
+
+    Paths where it is looked for executable is defined by 'PATH' environment
+    variable, 'os.confstr("CS_PATH")' or 'os.defpath'.
+
+    Args:
+        executable(str): Name of executable with or without extension. Can be
+            path to file.
+
+    Returns:
+        str: Full path to executable with extension (is file).
+        None: When the executable was not found.
+    """
+    if os.path.isfile(executable):
+        return executable
+
+    low_platform = platform.system().lower()
+    _, ext = os.path.splitext(executable)
+    variants = [executable]
+    if not ext:
+        if low_platform == "windows":
+            exts = [".exe", ".ps1", ".bat"]
+            for ext in os.getenv("PATHEXT", "").split(os.pathsep):
+                ext = ext.lower()
+                if ext and ext not in exts:
+                    exts.append(ext)
+        else:
+            exts = [".sh"]
+
+        for ext in exts:
+            variant = executable + ext
+            if os.path.isfile(variant):
+                return variant
+            variants.append(variant)
+
+    path_str = os.environ.get("PATH", None)
+    if path_str is None:
+        if hasattr(os, "confstr"):
+            path_str = os.confstr("CS_PATH")
+        elif hasattr(os, "defpath"):
+            path_str = os.defpath
+
+    if not path_str:
+        return None
+
+    paths = path_str.split(os.pathsep)
+    for path in paths:
+        for variant in variants:
+            filepath = os.path.abspath(os.path.join(path, executable))
+            if os.path.isfile(filepath):
+                return filepath
+    return None
```

Comment:
Out of curiosity. What is this logic based on? It looks like `distutils.find_executable` but does not [look like recent code](https://github.com/python/cpython/blob/main/Lib/distutils/spawn.py#L95-L129)?

Reply:
Source was distutils, but slightly modified. In distutils `find_executable` always expect `".exe"` on windows, this adds other options from `PATHEXT` and `.sh` for linux and mac.

Reply:
For reference sake this is what [`which`](https://github.com/getavalon/core/blob/master/avalon/lib.py#L103-L124) does in Avalon core which looks somewhat similar

Reply:
Handle less cases. When passed `executable` is existing path it should just return it. If the passed `executable` is path but don't have extension then this function will just add extension and return it. Then will start to look into PATH for possible variants.
Adding the `os.access(filepath, os.X_OK)` seems like a good idea.